xla/service/gpu/fusions/triton/triton_fusion_emitter_legacy_matmul.cc

```diff
@@ -1089,8 +1092,28 @@ class MatMulEmitterHelper {
     } else if (scope == TritonFusionAnalysis::Scope::RHS) {
       return dims_.rhs_noncontracting_dim_idx;
     } else {
-      CHECK(false) << "This shouldn't be called for the output scope.";
+      CHECK(false) << "This shouldn't be called for the other scopes.";
+    }
+  }
+
+  bool IsNonTrivialTiledDimension(TritonFusionAnalysis::Scope scope,
```

Comment:
nit: it would be slightly nicer to flip the check around to avoid the negation (i.e. make this `IsTrivialTiledDimension`)

Reply:
I don't think so because out of 4 combinations (trivial/non-trivial x tiled/non-tiled) we are only interested in this one: tiled and non-trivial.